Support shebang in .scala files with the Scala 3 dialect

## Configuration (required) ##
```
version = 3.5.8
runner.dialect = scala3
```

## Steps

Given code like this:
```scala
#!/usr/bin/env scala

@main def run() = println("hello world!")
```

## Problem

Scalafmt formats code like this:
```
.../main.scala:1: error: [dialect scala3] illegal start of definition identifier
#!/usr/bin/env scala
^
```

## Expectation

The file is legal scala3 (can be run with `scala main.scala` and compiled with `scalac main.scala`), and should be formatted while ignoring the first line.

## Notes

There seems to be supported for ammonite scripts, or rather anything with a `.sc` file ending

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala
@@ -6,7 +6,6 @@ import org.scalafmt.config.{ImportSelectors, Newlines, ScalafmtConfig, Spaces}
 import org.scalafmt.internal.ExpiresOn.{After, Before}
 import org.scalafmt.internal.Length.{Num, StateColumn}
 import org.scalafmt.internal.Policy.NoPolicy
-import org.scalafmt.sysops.FileOps
 import org.scalafmt.util._
 import org.scalameta.FileLine
 
@@ -89,8 +88,7 @@ class Router(formatOps: FormatOps) {
         Seq(Split(NoSplit.orNL(next(ft).right.is[T.EOF]), 0))
       case FormatToken(_: T.BOF, right, _) =>
         val policy = right match {
-          case T.Ident(name) // shebang in .sc files
-              if FileOps.isAmmonite(filename) && name.startsWith("#!") =>
+          case T.Ident(name) if name.startsWith("#!") =>
             val nl = findFirst(next(formatToken), Int.MaxValue) { x =>
               x.hasBreak || x.right.is[T.EOF]
             }

